src/allotropy/parsers/ctl_immunospot/ctl_immunospot_structure.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from dataclasses import dataclass
+import ntpath
+import re
+from typing import Optional
+
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.lines_reader import LinesReader
+from allotropy.parsers.utils.values import assert_not_none, try_float_or_none
+
+
+@dataclass
+class DeviceInfo:
+    device_identifier: str
+    model_number: str
+    equipment_serial_number: str
+    data_system_instance_id: str
+    basename: str
+    file_name: str
+    unc_path: str
+    software_name: str
+    software_version: str
+    measurement_time: str
+    analyst: str
+
+    @staticmethod
+    def create(reader: LinesReader) -> DeviceInfo:
+        data = {}
+        for i in range(3):  # for first three lines
+            line = assert_not_none(
+                reader.pop(),
+                msg=f"Unable to read line number {i+1}",
+            )
+            for element in line.split(";"):
+                parts = element.split(":", maxsplit=1)
+                if len(parts) > 1:
```

Comment:
Sanity checking: if this is not true - would we be missing expected data, or is that a valid use case. If we expect it to always be true, we should raise an error.

Reply:
The data that we want to capture is always evaluated as true with this condition. 
There might be other information present that is simply not needed and we want to ignore it.